### DISCUSSION TITLE
Cannot get zsh completions to work.

### DISCUSSION THREAD

**Discussion by asibahi** [eza-community/eza#364] [Q&A, Howto, Help]
Hello. 

I read https://github.com/eza-community/eza/issues/308 and followed the steps in the Readme.

I am on Mac OS Ventura. I installed eza through homebrew.

I added `export FPATH="/opt/homebrew/share/zsh/site-functions:$FPATH"` to my `.zshrc` file. This is the path where Homebrew tells me the shell completions are (when I type `brew_info`. When I was trying different solutions I also followed the steps outlined in the ReadMe by cloning the git repo and linking to that.

This is what I get when I `echo $fpath`.: `/opt/homebrew/share/zsh/site-functions /usr/local/share/zsh/site-functions /usr/share/zsh/site-functions /usr/share/zsh/5.9/functions` 

Neither the homebrew suggested folder nor the way outlined in the readme seem to work, in either Terminal or Warp. I managed to update the file and everything it just .. doesn't work.

What am I missing?

**Comment by cafkafk** [comment]
I'm not sure. I had the opportunity to check out the homebrew package on a MacOS laptop a few days ago, and I noticed that man pages also weren't present. I'm not sure how brew works under the hood, looking into it, it seems that they are trying to install them https://github.com/Homebrew/homebrew-core/blob/dcae06f64837ffb6df06e8795fb1c21a681f6f14/Formula/e/eza.rb#L24C1-L34C6.

This might be a setup issue. But I don't think it's an eza issue. Perhaps you could open this downstream, in https://github.com/Homebrew/homebrew-core?

**Comment by asibahi** [comment]
I used to have exa installed with brew and text completions worked then. 

I don't think it is a problem with eza really (hence the Question tag), but I'd like help debugging this. Note I also followed the steps in the eza README as well and they didn't work either. For all I know this could be a zsh issue.

**Comment by cafkafk** [comment]
I think this is better suited to a Q&A discussion then, I'll move it there.

**Comment by gierens** [comment]
Just tried this on my work macbook and can confirm that brew-installed stuff in general does not get completed.

I googled and fiddled around a little, and it seems like you need to initialize the completion after adding to `FPATH`. Adding the following to my `.zshrc` fixed it for me:
```zsh
if type brew &>/dev/null; then
    FPATH="$(brew --prefix)/share/zsh/site-functions:${FPATH}"
    autoload -Uz compinit
    compinit
fi
```
For reference:
- homebrew documentation: https://docs.brew.sh/Shell-Completion#configuring-completions-in-zsh on this
- related issue: https://github.com/Homebrew/brew/issues/8984

I made a PR to mention this in the README, see https://github.com/eza-community/eza/pull/468
